### PR TITLE
Bluetooth: controller: Add ll_adv_aux_set_remove function

### PR DESCRIPTION
**Pull request by dagbja**
Add function to remove aux advertising set.

Signed-off-by: Dag Bjarvin Dag.Bjarvin@nordicsemi.no

### CODE REVIEW

File: subsys/bluetooth/controller/ll_sw/nordic/lll/lll_adv.h [hunk 1]
```diff
@@ -121,6 +121,11 @@ static inline void lll_adv_pdu_enqueue(struct lll_adv_pdu *pdu, uint8_t idx)
 	pdu->last = idx;
 }
 
+static inline void lll_adv_pdu_dequeue(struct lll_adv_pdu *pdu)
```
Comment (cvinayak) [inline subsys/bluetooth/controller/ll_sw/nordic/lll/lll_adv.h:124]: I dont see these new functions being used.

File: subsys/bluetooth/controller/ll_sw/nordic/lll/lll_adv.h [hunk 2]
```diff
@@ -131,6 +136,11 @@ static inline void lll_adv_data_enqueue(struct lll_adv *lll, uint8_t idx)
 	lll_adv_pdu_enqueue(&lll->adv_data, idx);
 }
 
+static inline void lll_adv_data_dequeue(struct lll_adv *lll)
```
Comment (cvinayak) [inline subsys/bluetooth/controller/ll_sw/nordic/lll/lll_adv.h:139]: I dont see these new functions being used.

File: subsys/bluetooth/controller/ll_sw/nordic/lll/lll_adv.h [hunk 3]
```diff
@@ -165,6 +175,11 @@ static inline void lll_adv_aux_data_enqueue(struct lll_adv_aux *lll,
 	lll_adv_pdu_enqueue(&lll->data, idx);
 }
 
+static inline void lll_adv_aux_data_dequeue(struct lll_adv_aux *lll)
```
Comment (cvinayak) [inline subsys/bluetooth/controller/ll_sw/nordic/lll/lll_adv.h:178]: I dont see these new functions being used.

File: subsys/bluetooth/controller/ll_sw/ull_adv_aux.c [hunk 4]
```diff
@@ -527,8 +527,50 @@ uint8_t ll_adv_aux_set_count_get(void)
...
+		return BT_HCI_ERR_UNKNOWN_ADV_IDENTIFIER;
+	}
+
+	if (adv->is_enabled) {
+		return BT_HCI_ERR_CMD_DISALLOWED;
+	}
+
+	lll = &adv->lll;
+		/* Equivalent to lll_adv_data_dequeue(lll) */
+	lll->adv_data.first = lll->adv_data.last = 0;
```
Comment (cvinayak) [inline subsys/bluetooth/controller/ll_sw/ull_adv_aux.c:551]: This is not necessary. Instead add test for if new adv set can be created and that adv data is sane. Handle any issues during creation.

File: subsys/bluetooth/controller/ll_sw/ull_adv_aux.c [hunk 5]
```diff
@@ -527,8 +527,50 @@ uint8_t ll_adv_aux_set_count_get(void)
...
+
+#if defined(CONFIG_BT_CTLR_ADV_EXT)
```
Comment (cvinayak) [inline subsys/bluetooth/controller/ll_sw/ull_adv_aux.c:553]: conditional compile is redundant as this interface function is part of advertising extensions feature.

File: subsys/bluetooth/controller/ll_sw/ull_adv_aux.c [hunk 6]
```diff
@@ -527,8 +527,50 @@ uint8_t ll_adv_aux_set_count_get(void)
...
+	lll_aux = lll->aux;
+	/* Equivalent to lll_adv_aux_data_dequeue(lll_aux) */
+	lll_aux->data.first = lll_aux->data.last = 0;
```
Comment (cvinayak) [inline subsys/bluetooth/controller/ll_sw/ull_adv_aux.c]: This is not necessary.

File: subsys/bluetooth/controller/ll_sw/ull_adv_aux.c [hunk 7]
```diff
@@ -527,8 +527,50 @@ uint8_t ll_adv_aux_set_count_get(void)
...
+
+		/* Release auxiliary channel set */
+	if (lll_aux) {
+		struct ll_adv_aux_set* aux;
+		aux = (void*)HDR_LLL2EVT(lll_aux);
```
Comment (cvinayak) [inline subsys/bluetooth/controller/ll_sw/ull_adv_aux.c:561]: missing blank line.

File: subsys/bluetooth/controller/ll_sw/ull_adv_aux.c [hunk 8]
```diff
@@ -527,8 +527,50 @@ uint8_t ll_adv_aux_set_count_get(void)
...
+	/* Get the advertising set instance */
+	adv = ull_adv_is_created_get(handle);
+	if (!adv) {
+		return BT_HCI_ERR_UNKNOWN_ADV_IDENTIFIER;
+	}
+
+	if (adv->is_enabled) {
+		return BT_HCI_ERR_CMD_DISALLOWED;
+	}
+
```
Comment (cvinayak) [inline subsys/bluetooth/controller/ll_sw/ull_adv_aux.c:548]: Missing check, if periodic is created and enabled?

File: subsys/bluetooth/controller/ll_sw/ull_adv_aux.c [hunk 9]
```diff
@@ -527,8 +527,50 @@ uint8_t ll_adv_aux_set_count_get(void)
...
+	if (lll_aux) {
+		struct ll_adv_aux_set* aux;
+		aux = (void*)HDR_LLL2EVT(lll_aux);
+		ull_adv_aux_release(aux);
+	}
+#endif/* !CONFIG_BT_CTLR_ADV_EXT */
+
+#if defined(CONFIG_BT_CTLR_ADV_PERIODIC)
+	lll_sync = lll->sync;
+	lll_sync->data.first = lll_sync->data.last = 0;
```
Comment (cvinayak) [inline subsys/bluetooth/controller/ll_sw/ull_adv_aux.c:568]: This is not necessary.

File: subsys/bluetooth/controller/ll_sw/ull_adv_aux.c [hunk 10]
```diff
@@ -527,8 +527,50 @@ uint8_t ll_adv_aux_set_count_get(void)
...
+
+	/* need to call sync_release? */
```
Comment (cvinayak) [inline subsys/bluetooth/controller/ll_sw/ull_adv_aux.c]: Add a `/* TODO: */` or `/* FIXME: */` comment if implementing in future.

File: tests/bluetooth/bsim_bt/bsim_test_advx/src/main.c [hunk 11]
```diff
@@ -188,6 +188,15 @@ static void test_advx_main(void)
...
 
+	k_sleep(K_MSEC(400));
+
+	printk("Removing adv aux set...");
+	err = ll_adv_aux_set_remove(handle);
+	if (err) {
+		goto exit;
+	}
+	printk("success.\n");
+
```
Comment (cvinayak) [inline tests/bluetooth/bsim_bt/bsim_test_advx/src/main.c:199]: Missing tests to check if adv set can be created and used.

Is the bsim test passing? does line 194, 200, and 206 succeed after adv set is removed?

**Review by cvinayak — CHANGES_REQUESTED** [review]
Why do you have 3 commits with same commit message?

File: subsys/bluetooth/controller/ll_sw/ull_adv_aux.c [hunk 12]
```diff
@@ -527,8 +527,50 @@ uint8_t ll_adv_aux_set_count_get(void)
...
+		return BT_HCI_ERR_UNKNOWN_ADV_IDENTIFIER;
+	}
+
+	if (adv->is_enabled) {
+		return BT_HCI_ERR_CMD_DISALLOWED;
+	}
+
+	lll = &adv->lll;
+		/* Equivalent to lll_adv_data_dequeue(lll) */
+	lll->adv_data.first = lll->adv_data.last = 0;
```
Comment (kruithofa) [inline subsys/bluetooth/controller/ll_sw/ull_adv_aux.c:551]: Note also that the assignment violates https://docs.zephyrproject.org/latest/guides/coding_guidelines/index.html
rule 13.4

File: subsys/bluetooth/controller/ll_sw/nordic/lll/lll_adv.h (hunk 1 above)
Comment (dagbja) [reply subsys/bluetooth/controller/ll_sw/nordic/lll/lll_adv.h]: Removed.

File: subsys/bluetooth/controller/ll_sw/nordic/lll/lll_adv.h (hunk 2 above)
Comment (dagbja) [reply subsys/bluetooth/controller/ll_sw/nordic/lll/lll_adv.h]: Removed.

File: subsys/bluetooth/controller/ll_sw/nordic/lll/lll_adv.h (hunk 3 above)
Comment (dagbja) [reply subsys/bluetooth/controller/ll_sw/nordic/lll/lll_adv.h]: Removed.

File: subsys/bluetooth/controller/ll_sw/ull_adv_aux.c (hunk 5 above)
Comment (dagbja) [reply subsys/bluetooth/controller/ll_sw/ull_adv_aux.c]: Fixed.

File: subsys/bluetooth/controller/ll_sw/ull_adv_aux.c (hunk 7 above)
Comment (dagbja) [reply subsys/bluetooth/controller/ll_sw/ull_adv_aux.c:554]: Fixed.

File: subsys/bluetooth/controller/ll_sw/ull_adv_aux.c (hunk 9 above)
Comment (dagbja) [reply subsys/bluetooth/controller/ll_sw/ull_adv_aux.c]: Removed.

File: tests/bluetooth/bsim_bt/bsim_test_advx/src/main.c (hunk 11 above)
Comment (dagbja) [reply tests/bluetooth/bsim_bt/bsim_test_advx/src/main.c]: Modified bsim test. It is passing.

File: subsys/bluetooth/controller/ll_sw/ull_adv_aux.c (hunk 8 above)
Comment (dagbja) [reply subsys/bluetooth/controller/ll_sw/ull_adv_aux.c:548]: Added

File: subsys/bluetooth/controller/ll_sw/ull_adv_aux.c (hunk 10 above)
Comment (dagbja) [reply subsys/bluetooth/controller/ll_sw/ull_adv_aux.c]: Added

File: subsys/bluetooth/controller/ll_sw/ull_adv_aux.c (hunk 12 above)
Comment (dagbja) [reply subsys/bluetooth/controller/ll_sw/ull_adv_aux.c]: Fixed

File: subsys/bluetooth/controller/ll_sw/ull_adv_aux.c (hunk 4 above)
Comment (dagbja) [reply subsys/bluetooth/controller/ll_sw/ull_adv_aux.c]: Removed.

File: subsys/bluetooth/controller/ll_sw/ull_adv_aux.c [hunk 13]
```diff
@@ -527,8 +527,52 @@ uint8_t ll_adv_aux_set_count_get(void)
...
 {
-	/* TODO: reset/release primary channel and Aux channel PDUs */
-	return 0;
+	struct ll_adv_set *adv;
+	struct lll_adv *lll;
+#if defined(CONFIG_BT_CTLR_ADV_EXT)
+	struct lll_adv_aux *lll_aux;
+#endif/* !CONFIG_BT_CTLR_ADV_EXT */
+
+		/* Get the advertising set instance */
```
Comment (cvinayak) [inline subsys/bluetooth/controller/ll_sw/ull_adv_aux.c:536]: Redundant leading whitespace

File: subsys/bluetooth/controller/ll_sw/ull_adv_aux.c [hunk 14]
```diff
@@ -527,8 +527,52 @@ uint8_t ll_adv_aux_set_count_get(void)
...
+		lll_sync = lll->sync;
+		sync = (void *)HDR_LLL2EVT(lll_sync);
+
+		if (sync->is_enabled) {
+			return BT_HCI_ERR_CMD_DISALLOWED;
+		}
+	}
+#endif /* CONFIG_BT_CTLR_ADV_PERIODIC */
+
+	lll = &adv->lll;
```
Comment (cvinayak) [inline subsys/bluetooth/controller/ll_sw/ull_adv_aux.c:561]: Redundant assignment if line 547 is compiled in. Move to line above 546 and remove from here.

File: subsys/bluetooth/controller/ll_sw/ull_adv_aux.c [hunk 15]
```diff
@@ -527,8 +527,52 @@ uint8_t ll_adv_aux_set_count_get(void)
...
+		return BT_HCI_ERR_CMD_DISALLOWED;
+	}
+
+#if defined(CONFIG_BT_CTLR_ADV_PERIODIC)
+	lll = &adv->lll;
+	if (lll->sync) {
+		struct lll_adv_sync *lll_sync;
+		struct ll_adv_sync_set *sync;
+
+		lll_sync = lll->sync;
```
Comment (cvinayak) [inline subsys/bluetooth/controller/ll_sw/ull_adv_aux.c:552]: No need to have an auto variable as de-referencing only once in the next line.

File: subsys/bluetooth/controller/ll_sw/ull_adv_aux.c [hunk 16]
```diff
@@ -527,8 +527,52 @@ uint8_t ll_adv_aux_set_count_get(void)
 
 uint8_t ll_adv_aux_set_remove(uint8_t handle)
 {
-	/* TODO: reset/release primary channel and Aux channel PDUs */
-	return 0;
+	struct ll_adv_set *adv;
+	struct lll_adv *lll;
+#if defined(CONFIG_BT_CTLR_ADV_EXT)
```
Comment (cvinayak) [inline subsys/bluetooth/controller/ll_sw/ull_adv_aux.c:532]: Redundant conditional compilation. The whole file is included only for advertising extensions support.

File: subsys/bluetooth/controller/ll_sw/ull_adv_aux.c [hunk 17]
```diff
@@ -527,8 +527,52 @@ uint8_t ll_adv_aux_set_count_get(void)
...
+		/* Release auxiliary channel set */
+	if (lll_aux) {
+		struct ll_adv_aux_set *aux;
+
+		aux = (void *)HDR_LLL2EVT(lll_aux);
+
+		ull_adv_aux_release(aux);
+	}
+
+	/* TODO: Handle sync if implemented in future. */
```
Comment (cvinayak) [inline subsys/bluetooth/controller/ll_sw/ull_adv_aux.c:573]: Remove this comment.

File: tests/bluetooth/bsim_bt/bsim_test_advx/src/main.c [hunk 18]
```diff
@@ -247,6 +247,83 @@ static void test_advx_main(void)
...
+	k_sleep(K_MSEC(1000));
+
+	printk("enabling...");
+	err = ll_adv_enable(handle, 1);
```
Comment (cvinayak) [inline tests/bluetooth/bsim_bt/bsim_test_advx/src/main.c:296]: The expected result should be `BT_HCI_ERR_UNKNOWN_ADV_IDENTIFIER`

File: subsys/bluetooth/controller/ll_sw/ull_adv_aux.c [hunk 19]
```diff
@@ -527,8 +527,52 @@ uint8_t ll_adv_aux_set_count_get(void)
...
+
```
Comment (cvinayak) [inline subsys/bluetooth/controller/ll_sw/ull_adv_aux.c:574]: You are missing release of the advertising set by unset of is_created.

File: tests/bluetooth/bsim_bt/bsim_test_advx/src/main.c [hunk 20]
```diff
@@ -247,6 +247,83 @@ static void test_advx_main(void)
...
+	printk("Update advertising data 2...");
+	err = ll_adv_aux_ad_data_set(handle, AD_OP, AD_FRAG_PREF,
+				     sizeof(adv_data2), (void *)adv_data2);
+	if (err) {
+		goto exit;
+	}
+	printk("success.\n");
+
+	k_sleep(K_MSEC(400));
+
```
Comment (cvinayak) [inline tests/bluetooth/bsim_bt/bsim_test_advx/src/main.c:278]: These are un-related to adv set remove, remove it. Only test by creating, enabling and removing; creating and removing; removing without creating etc.

```suggestion

```

File: tests/bluetooth/bsim_bt/bsim_test_advx/src/main.c [hunk 21]
```diff
@@ -247,6 +247,83 @@ static void test_advx_main(void)
...
+	err = ll_adv_enable(handle, 1);
+	if (err) {
+		goto exit;
+	}
+	printk("success.\n");
+
+	k_sleep(K_MSEC(400));
+
+	printk("Update advertising data 1...");
+	err = ll_adv_aux_ad_data_set(handle, AD_OP, AD_FRAG_PREF,
```
Comment (cvinayak) [inline tests/bluetooth/bsim_bt/bsim_test_advx/src/main.c:305]: Keep this, for testing auxiliary set instantiation, and then testing the removal.

**Review by cvinayak — CHANGES_REQUESTED** [review]

File: tests/bluetooth/bsim_bt/bsim_test_advx/src/main.c [hunk 22]
```diff
@@ -247,6 +247,83 @@ static void test_advx_main(void)
 	}
 	printk("success.\n");
 
+	printk("enabling...");
```
Comment (kruithofa) [inline tests/bluetooth/bsim_bt/bsim_test_advx/src/main.c:250]: For all tests: this only tests that the HCI commands are working as intended. Is it possible to add tests for the actual functionality behind the HCI commands?

File: subsys/bluetooth/controller/ll_sw/ull_adv_aux.c [hunk 23]
```diff
@@ -527,8 +527,52 @@ uint8_t ll_adv_aux_set_count_get(void)
...
+		if (sync->is_enabled) {
+			return BT_HCI_ERR_CMD_DISALLOWED;
+		}
+	}
+#endif /* CONFIG_BT_CTLR_ADV_PERIODIC */
+
+	lll = &adv->lll;
+	lll_aux = lll->aux;
+
+		/* Release auxiliary channel set */
```
Comment (kruithofa) [inline subsys/bluetooth/controller/ll_sw/ull_adv_aux.c:564]: Indentation

**Review by kruithofa — CHANGES_REQUESTED** [review]

File: subsys/bluetooth/controller/ll_sw/ull_adv_aux.c (hunk 13 above)
Comment (dagbja) [reply subsys/bluetooth/controller/ll_sw/ull_adv_aux.c:534]: Fixed

File: subsys/bluetooth/controller/ll_sw/ull_adv_aux.c (hunk 14 above)
Comment (dagbja) [reply subsys/bluetooth/controller/ll_sw/ull_adv_aux.c]: Fixed

File: subsys/bluetooth/controller/ll_sw/ull_adv_aux.c (hunk 15 above)
Comment (dagbja) [reply subsys/bluetooth/controller/ll_sw/ull_adv_aux.c]: Fixed

File: subsys/bluetooth/controller/ll_sw/ull_adv_aux.c (hunk 16 above)
Comment (dagbja) [reply subsys/bluetooth/controller/ll_sw/ull_adv_aux.c]: Fixed

File: subsys/bluetooth/controller/ll_sw/ull_adv_aux.c (hunk 17 above)
Comment (dagbja) [reply subsys/bluetooth/controller/ll_sw/ull_adv_aux.c]: Removed

File: subsys/bluetooth/controller/ll_sw/ull_adv_aux.c (hunk 19 above)
Comment (dagbja) [reply subsys/bluetooth/controller/ll_sw/ull_adv_aux.c:570]: Fixed

File: subsys/bluetooth/controller/ll_sw/ull_adv_aux.c (hunk 23 above)
Comment (dagbja) [reply subsys/bluetooth/controller/ll_sw/ull_adv_aux.c:560]: Not clear on what you mean. We want 1 tab indentation on comments, right?

Comment (kruithofa) [reply subsys/bluetooth/controller/ll_sw/ull_adv_aux.c:560]: Indentation should be the same as the code around it

Comment (dagbja) [reply subsys/bluetooth/controller/ll_sw/ull_adv_aux.c:560]: Is that new? I believe I've had review comments in the past expecting the comment to be indented one tab more than following code. (I might be wrong.)

Comment (dagbja) [reply subsys/bluetooth/controller/ll_sw/ull_adv_aux.c:560]: But I see other comments are indented like you said, so I'll fix it.

File: tests/bluetooth/bsim_bt/bsim_test_advx/src/main.c (hunk 18 above)
Comment (dagbja) [reply tests/bluetooth/bsim_bt/bsim_test_advx/src/main.c:287]: Changed the tests.

File: tests/bluetooth/bsim_bt/bsim_test_advx/src/main.c (hunk 20 above)
Comment (dagbja) [reply tests/bluetooth/bsim_bt/bsim_test_advx/src/main.c]: Changed the tests .

File: tests/bluetooth/bsim_bt/bsim_test_advx/src/main.c (hunk 21 above)
Comment (dagbja) [reply tests/bluetooth/bsim_bt/bsim_test_advx/src/main.c]: Done.

File: subsys/bluetooth/controller/ll_sw/ull_adv_aux.c (hunk 23 above)
Comment (dagbja) [reply subsys/bluetooth/controller/ll_sw/ull_adv_aux.c]: Fixed.

File: tests/bluetooth/bsim_bt/bsim_test_advx/src/main.c (hunk 22 above)
Comment (dagbja) [reply tests/bluetooth/bsim_bt/bsim_test_advx/src/main.c]: The old tests use HCI commands. The new tests use link layer calls. I don't think we can easily get more detailed than that.

File: tests/bluetooth/bsim_bt/bsim_test_advx/src/main.c [hunk 24]
```diff
@@ -238,7 +238,7 @@ static void test_advx_main(void)
 	}
 	printk("success.\n");
 
-	k_sleep(K_MSEC(400));
+	k_sleep(K_MSEC(1000));
```
Comment (cvinayak) [inline tests/bluetooth/bsim_bt/bsim_test_advx/src/main.c:241]: Keep this 400 millisecond. (Sufficient to analyse periodic adv PDUs in imported Ellisys traces).

File: tests/bluetooth/bsim_bt/bsim_test_advx/src/main.c [hunk 25]
```diff
@@ -247,6 +247,80 @@ static void test_advx_main(void)
 	}
 	printk("success.\n");
 
+	k_sleep(K_MSEC(400));
+
+	printk("Removing adv aux set that's created and disabled ...");
+	err = ll_adv_aux_set_remove(handle);
+	if (err) {
+		printk("err = %d\n", err);
```
Comment (cvinayak) [inline tests/bluetooth/bsim_bt/bsim_test_advx/src/main.c:255]: All the printk's are redundant, as err is printed in line 330.

File: tests/bluetooth/bsim_bt/bsim_test_advx/src/main.c [hunk 26]
```diff
@@ -247,6 +247,80 @@ static void test_advx_main(void)
 	}
 	printk("success.\n");
 
+	k_sleep(K_MSEC(400));
```
Comment (cvinayak) [inline tests/bluetooth/bsim_bt/bsim_test_advx/src/main.c:250]: Do not need delays when there is no on-air activity that needs analysis.

File: tests/bluetooth/bsim_bt/bsim_test_advx/src/main.c [hunk 27]
```diff
@@ -247,6 +247,80 @@ static void test_advx_main(void)
...
+
+	printk("Disabling adv set...");
+	err = ll_adv_enable(handle, 0);
+	if (err) {
+		printk("err = %d\n", err);
+		goto exit;
+	}
+	printk("success.\n");
+
+	k_sleep(K_MSEC(1000));
```
Comment (cvinayak) [inline tests/bluetooth/bsim_bt/bsim_test_advx/src/main.c:314]: Do not need this delay.

**Review by cvinayak — CHANGES_REQUESTED** [review]
Code now looks good. Some minor test change to be taken care of. I need to test this (alongwith my other local changes, pending).

**Comment by cvinayak** [comment]
Please separate out into 3 commits:
1. controller changes.
2. shell application change
3. bsim test change.

**Comment by dagbja** [comment]
First line of the commit messages:

Bluetooth: controller: Add ll_adv_aux_set_remove function

For shell app and bsim, should I write something other than “controller”?

**Comment by dagbja** [comment]
I used

Bluetooth: tests:

based on an example from Vinayak’s branch.

From: Bjarvin, Dag
Sent: Friday, July 24, 2020 7:35 AM
To: 'zephyrproject-rtos/zephyr' <reply@reply.github.com>; zephyrproject-rtos/zephyr <zephyr@noreply.github.com>
Cc: Author <author@noreply.github.com>
Subject: RE: [zephyrproject-rtos/zephyr] Adv aux set remove (#26918)

First line of the commit messages:

Bluetooth: controller: Add ll_adv_aux_set_remove function

For shell app and bsim, should I write something other than “controller”?

File: tests/bluetooth/bsim_bt/bsim_test_advx/src/main.c (hunk 25 above)
Comment (dagbja) [reply tests/bluetooth/bsim_bt/bsim_test_advx/src/main.c:255]: Oh, sorry, I added those during my testing. Will remove.

File: tests/bluetooth/bsim_bt/bsim_test_advx/src/main.c (hunk 24 above)
Comment (dagbja) [reply tests/bluetooth/bsim_bt/bsim_test_advx/src/main.c]: Fixed.

File: tests/bluetooth/bsim_bt/bsim_test_advx/src/main.c (hunk 25 above)
Comment (dagbja) [reply tests/bluetooth/bsim_bt/bsim_test_advx/src/main.c]: Fixed.

File: tests/bluetooth/bsim_bt/bsim_test_advx/src/main.c (hunk 26 above)
Comment (dagbja) [reply tests/bluetooth/bsim_bt/bsim_test_advx/src/main.c]: Fixed.

File: tests/bluetooth/bsim_bt/bsim_test_advx/src/main.c (hunk 27 above)
Comment (dagbja) [reply tests/bluetooth/bsim_bt/bsim_test_advx/src/main.c]: Fixed.

**Review by cvinayak — CHANGES_REQUESTED** [review]

File: subsys/bluetooth/controller/ll_sw/ull_adv_aux.c [hunk 28]
```diff
@@ -527,8 +527,48 @@ uint8_t ll_adv_aux_set_count_get(void)
 
 uint8_t ll_adv_aux_set_remove(uint8_t handle)
 {
-	/* TODO: reset/release primary channel and Aux channel PDUs */
-	return 0;
+	struct ll_adv_set *adv;
+	struct lll_adv *lll;
+	struct lll_adv_aux *lll_aux;
```
[the remark this answers is not in the document]
Comment (dagbja) [reply subsys/bluetooth/controller/ll_sw/ull_adv_aux.c]: Fixed.

File: subsys/bluetooth/shell/ll.c [hunk 29]
```diff
@@ -275,6 +275,10 @@ int cmd_advx(const struct shell *shell, size_t argc, char *argv[])
 	err = ll_adv_enable(handle, enable, 0, 0, 0, 0, 0);
 #else /* !CONFIG_BT_HCI_MESH_EXT */
 	err = ll_adv_enable(handle, enable);
+
```
[the remark this answers is not in the document]
Comment (dagbja) [reply subsys/bluetooth/shell/ll.c]: Fixed.

File: tests/bluetooth/bsim_bt/bsim_test_advx/src/main.c (hunk 22 above)
Comment (kruithofa) [reply tests/bluetooth/bsim_bt/bsim_test_advx/src/main.c]: My point is that you've tested that the HCI interface is working, but how do we know that the adv aux set really was removed (i.e. that the function ll_adv_aux_set_remove works as intended)

**Review by cvinayak — APPROVED** [review]

File: tests/bluetooth/bsim_bt/bsim_test_advx/src/main.c (hunk 22 above)
Comment (dagbja) [reply tests/bluetooth/bsim_bt/bsim_test_advx/src/main.c]: I guess we can make a second call to ll_adv_aux_set_remove. If the set is removed (is_created=0), we should expect the call to fail with BT_HCI_ERR_UNKNOWN_ADV_IDENTIFIER.

Comment (dagbja) [reply tests/bluetooth/bsim_bt/bsim_test_advx/src/main.c]: Actually, we alreadyhave this test - "Removing adv aux set that's not created".

**Review by kruithofa — APPROVED** [review]

**Review by carlescufi — APPROVED** [review]

**Pull request merged**